torch/distributed/rpc/__init__.py

```diff
@@ -195,23 +201,48 @@ def _validate_rpc_args(backend, store, name, rank, world_size, rpc_backend_optio
             store: dist.Store,
             name: str,
             rank: numbers.Integral,
-            world_size: numbers.Integral,
+            # world_size can be None for a dynamic group
+            world_size: (numbers.Integral, type(None)),
             rpc_backend_options: RpcBackendOptions,
         }
         for arg, arg_type in type_mapping.items():
-            if not isinstance(arg, arg_type):
+            if not isinstance(arg, arg_type):  # type: ignore[arg-type]
                 raise RuntimeError(
                     "Argument {} must be of type {} but got type {}".format(
                         arg, arg_type, type(arg)
                     )
                 )
 
+    def _create_store_for_rpc(backend_options, rank):
+        def _get_env_or_raise(env_var: str) -> str:
+            env_val = os.environ.get(env_var, None)
+            if not env_val:
+                raise ValueError("environment variable %s expected, but not set" % env_var)
+            else:
+                return env_val
+
+        result = urlparse(backend_options.init_method)
+        scheme = result.scheme
+        query = dict(pair.split("=") for pair
+                     in filter(None, result.query.split("&")))  # type: ignore[misc, arg-type, var-annotated]
+        if scheme == "file":
+            ValueError("File initialization is not supported for Dynamic RPC")
+        else:
+            hostname = _get_env_or_raise("MASTER_ADDR")
+            port = int(_get_env_or_raise("MASTER_PORT"))
+        if rank == -1:
+            if "rank" in query:
+                rank = int(query["rank"])
+            else:
+                rank = int(_get_env_or_raise("RANK"))
+        return _create_c10d_store(hostname, port, rank, None, backend_options.rpc_timeout)
```

Comment:
Can we consolidate duplicated code with `rendezvous()`? E.g., we can have a helper function that wraps all the common code and the both this function and `rendezvous()` call into that function.

Reply:
Thanks! I merged this method to also use the rendezvous handlers